### DISCUSSION TITLE
Add the ability to have a Quiz or a Test

### DISCUSSION THREAD

**Discussion by aaronlippold** [vuepress-theme-hope/vuepress-theme-hope#3590] [Ideas]
For those using the theme for student content, having the ability to have a quiz or test would be very useful

**Comment by Mister-Hope** [comment]
I think this could be a component written by yourself. In another word, I don't think I should do anything.

The only way I can think out, is something like this:

```md
::: test

@question Question 1

What's the color of apple?

@option:correct A

red

@option B

blue

@option C

green

@option D

grey

:::
```

Any detailed ideas about what you are talking? A markdown syntax above or anything else?


Or maybe:

```md
::: test

Question content...

Can be multi line

@A xxxxx
@B xxxxx

   xxxx

@C xxxxx
@D xxxxx

@answer ABD

:::
```

**Reply by aaronlippold** [reply]
I have seen a few ways folks have done a JSON like structure. I can look for a vue plugin to see if we can find some examples. The theme has been great for my student classes and I am now hoping to add a 'check your learning' section. 

Perhaps a component approach could work here as well.

**Comment by aaronlippold** [comment]
There are some example markdown quiz generators I found

https://github.com/bonartm/quizdown-js

https://github.com/osandadeshan/markdown-quiz-generator

Which could be example of possible markdown syntax - but I am sure Google has a few others out there
